packages/core/src/__tests__/provider.test.ts

```diff
@@ -0,0 +1,136 @@
+import type { Transaction } from "viem";
+import { polygonMumbai } from "viem/chains";
+import { afterEach, beforeEach, describe, it, vi } from "vitest";
+import { SmartAccountProvider } from "../provider/base.js";
+import type { UserOperationReceipt } from "../types.js";
+
+describe(
+  "Base Tests",
+  () => {
+    let retryMsDelays: number[] = [];
+
+    beforeEach(() => {
+      vi.useFakeTimers();
+      vi.spyOn(global, "setTimeout").mockImplementation(
+        // @ts-ignore: Mock implementation doesn't need to return a Timeout.
+        (callback: () => void, ms) => {
+          if (ms != null) {
+            retryMsDelays.push(ms);
+          }
+          callback();
+        }
+      );
+    });
+
+    afterEach(() => {
+      vi.restoreAllMocks();
+      retryMsDelays = [];
+    });
+
+    it("should apply only the initial delay for waitForUserOperationTransaction", async () => {
+      const providerMock = new SmartAccountProvider(
+        "ALCHEMY_RPC_URL",
+        "0xENTRYPOINT_ADDRESS",
+        polygonMumbai
+      );
+
+      const getUserOperationReceiptMock = vi
+        .spyOn(providerMock, "getUserOperationReceipt")
+        .mockImplementation(() => {
+          return Promise.resolve({
+            receipt: { transactionHash: "0xMOCK_USER_OP_RECEIPT" },
+          } as unknown as UserOperationReceipt);
```

Comment:
I casted to `unknown` to avoid ballooning the mocked return value with irrelevant fields for the tested code path. lmk if there's a better way!